include/copc-lib/io/reader.hpp

```diff
@@ -4,16 +4,49 @@
 #include <istream>
 
 #include <copc-lib/copc/file.hpp>
+#include <copc-lib/hierarchy/internal/hierarchy.hpp>
+#include <copc-lib/hierarchy/node.hpp>
+#include <copc-lib/hierarchy/page.hpp>
+#include <copc-lib/io/base_io.hpp>
+#include <copc-lib/las/vlr.hpp>
 
-namespace copc::io
+namespace copc
 {
-class Reader
+class Reader : public BaseIO
 {
   public:
     Reader(std::istream &in_stream);
 
-    std::shared_ptr<CopcFile> file;
+    // Reads the node's data into an uncompressed byte array
+    std::vector<char> GetPointData(Node const &node);
```

Comment:
Let's add a `std::vector<char> GetPointData(VoxelKey const &key);` that does all the proper checks and maybe return an empty vector if key doesn't exist?

Reply:
More of an easy function to use, not requiring a try/catch kinda setup.